django/template/context.py

```diff
@@ -268,6 +268,9 @@ def make_context(context, request=None, **kwargs):
     """
     Create a suitable Context from a plain dict and optionally an HttpRequest.
     """
+    if context is not None:
```

Comment:
Instead of this i propose something along the lines of.
```python
if context is not None:
        if not isinstance(context, RequestContext) and not isinstance(context, Context) and not isinstance(context, dict):
            raise TypeError('Context must be a dict rather than %r.' % context.__class__.__name__)
    if request is None:
        if context is not None:
            if isinstance(context, RequestContext):
                raise TypeError('RequestContext cannot be passed without RequestFactory Object')
        context = Context(context, **kwargs)
```